File: plant-back/SharedLibrary/Models/GenericResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SharedLibrary.Models
{
    public class GenericResponse<T>
    {
        public GenericResponse()
        {
        }

        public GenericResponse(T data)
        {
            Status = ResponseStatus.OK;
            Message = "Success";
            Data = data;
        }

        public GenericResponse(ResponseStatus status, string message, T data)
        {
            Status = status;
            Message = message;
            Data = data;
        }

        public ResponseStatus Status { get; set; }
        public string Message { get; set; }

        public T Data { get; set; }
    }

    public enum ResponseStatus
    {
        OK,
        Warning,
        Error
    }
}

File: plant-front/src/app/core/models/generic-response.ts
import { ResponseStatus } from "../enums/response-status";

export interface IGenericResponse<T> {
  status: ResponseStatus;
  message: string;
  data: T;
}

export class GenericResponse<T> {

  public Status: ResponseStatus;
  public Message: string;
  public Data: T;

  constructor(json: IGenericResponse<T>) {
    this.Status = json.status;
    this.Message = json.message;
    this.Data = json.data;
	}
}




